libp2p/src/main/kotlin/io/libp2p/mux/yamux/YamuxHandler.kt

```diff
@@ -240,8 +245,11 @@ open class YamuxHandler(
         streamHandlers.remove(child.id)?.dispose()
     }
 
-    private fun calculateTotalBufferedWrites(): Int {
-        return streamHandlers.values.sumOf { it.sendBuffer.readableBytes() }
+    override fun checkCanOpenNewStream() {
```

Comment:
Why not just throw from `onLocalOpen()` ?

Reply:
So essentially in `newStream` method in `AbstractMuxHandler`, if `onLocalOpen` fails then `outboundInitializer` is never called and `createStream` in `MuxHandler` controller future is never complete. And I am thinking also in order to avoid calling `createChild` with he new id.

Reply:
With the fix above throwing from `onLocalOpen()` worked for me

Reply:
Nice. I like it.